src/reanimated2/Hooks.js

```diff
@@ -423,6 +431,13 @@ export function useAnimatedGestureHandler(handlers, dependencies) {
       savedDependencies: [],
     };
   }
+
+  useEffect(()=>{
+    return () => {
+      initRef.current = null;
```

Comment:
same ads above, couldn't we just quit using refs in such a case?

Reply:
No, we couldn't. It will be executed only when the view unmounts.